module/extension/Director/src/director/run_task_pack.cpp

```diff
@@ -229,10 +229,50 @@ namespace module::extension {
         group.done = false;
         group.update_data();
 
+        // Check if a Wait command was emitted and schedule to run the Provider again
+        // Other tasks can run with Wait
+        // Wait should be removed and then readded at the end
+        TaskList non_wait_tasks;
+        TaskList wait_tasks;
+        for (const auto& t : requested_tasks) {
```

Comment:
If a wait is optional or not optional should it happen with the other tasks? Or is wait special enough that it ignore that. Probably always expect wait to wait?

Reply:
as in, do we want it to only run when all subtasks can run? I think that could maybe be good, but could be something to add in later if someone wants that kind of functionality.